gateway/ln-gateway/src/gateway_module_v2/mod.rs

```diff
@@ -123,8 +123,8 @@ impl ClientModule for GatewayClientModuleV2 {
 
     fn context(&self) -> Self::ModuleStateMachineContext {
         GatewayClientContextV2 {
+            module: self.clone(),
             decoder: self.decoder(),
-            notifier: self.notifier.clone(),
             tpe_agg_pk: self.cfg.tpe_agg_pk,
             tpe_pks: self.cfg.tpe_pks.clone(),
             gateway: self.gateway.clone(),
```

Comment:
Do we need the whole module here? If you do, it actually makes the rest of the fields redundant.

Reply:
We ant to call the await_receive method in the Complete state machine.

Reply:
Discussed this offline, we both agree there is a bit of a code smell here, but its not worth the refactor in this PR.